<?xml version="1.0" encoding="UTF-8"?>
<testsuite name="MusicAdvisorTest" tests="1" skipped="0" failures="1" errors="0" timestamp="2021-05-16T08:35:43" hostname="Christians-MacBook-Pro.local" time="51.096">
  <properties/>
  <testcase name="start" classname="MusicAdvisorTest" time="51.096">
    <failure message="java.lang.AssertionError: Exception in test #8&#10;&#10;java.lang.IllegalStateException: Not a JSON Object: null&#10;&#9;at com.google.gson.JsonElement.getAsJsonObject(JsonElement.java:91)&#10;&#9;at advisor.GetPlaylist.getPlaylist(GetPlaylist.java:58)&#10;&#9;at advisor.Controller.showMenu(Controller.java:38)&#10;&#9;at advisor.Controller.startProgram(Controller.java:10)&#10;&#9;at advisor.Main.main(Main.java:17)&#10;&#10;Please find below the output of your program during this failed test.&#10;Note that the '&gt;' character indicates the beginning of the input line.&#10;&#10;---&#10;&#10;Arguments: -access http://127.0.0.1:45678 -resource http://127.0.0.1:56789&#10;&#10;&gt; auth&#10;use this link to request the access code:&#10;http://127.0.0.1:45678/authorize?client_id=fdea9ae7b29147fa98559b2030212918&amp;redirect_uri=http://localhost:8080&amp;response_type=code&#10;waiting for code...&#10;Tester: making requests to redirect uri: http://localhost:8080&#10;Tester: Error: Connection refused&#10;making http request for access_token...&#10;response:&#10;{&quot;access_token&quot;:&quot;456456&quot;,&quot;token_type&quot;:&quot;Bearer&quot;,&quot;expires_in&quot;:3600,&quot;refresh_token&quot;:&quot;567567&quot;,&quot;scope&quot;:&quot;&quot;}&#10;&#10;---SUCCESS---&#10;&gt; playlists Party Time" type="java.lang.AssertionError">java.lang.AssertionError: Exception in test #8

java.lang.IllegalStateException: Not a JSON Object: null
	at com.google.gson.JsonElement.getAsJsonObject(JsonElement.java:91)
	at advisor.GetPlaylist.getPlaylist(GetPlaylist.java:58)
	at advisor.Controller.showMenu(Controller.java:38)
	at advisor.Controller.startProgram(Controller.java:10)
	at advisor.Main.main(Main.java:17)

Please find below the output of your program during this failed test.
Note that the '&gt;' character indicates the beginning of the input line.

---

Arguments: -access http://127.0.0.1:45678 -resource http://127.0.0.1:56789

&gt; auth
use this link to request the access code:
http://127.0.0.1:45678/authorize?client_id=fdea9ae7b29147fa98559b2030212918&amp;redirect_uri=http://localhost:8080&amp;response_type=code
waiting for code...
Tester: making requests to redirect uri: http://localhost:8080
Tester: Error: Connection refused
making http request for access_token...
response:
{&quot;access_token&quot;:&quot;456456&quot;,&quot;token_type&quot;:&quot;Bearer&quot;,&quot;expires_in&quot;:3600,&quot;refresh_token&quot;:&quot;567567&quot;,&quot;scope&quot;:&quot;&quot;}

---SUCCESS---
&gt; playlists Party Time
	at org.junit.Assert.fail(Assert.java:88)
	at org.hyperskill.hstest.stage.StageTest.start(StageTest.java:141)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:119)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.base/java.lang.Thread.run(Thread.java:834)
</failure>
  </testcase>
  <system-out><![CDATA[?[1;31m
Start test 1?[0m
> auth
use this link to request the access code:
http://127.0.0.1:45678/authorize?client_id=fdea9ae7b29147fa98559b2030212918&redirect_uri=http://localhost:8080&response_type=code
waiting for code...
Tester: making requests to redirect uri: http://localhost:8080
Tester: Error: Connection refused
making http request for access_token...
response:
{"access_token":"456456","token_type":"Bearer","expires_in":3600,"refresh_token":"567567","scope":""}

---SUCCESS---
> exit
---GOODBYE!---
?[1;31m
Start test 2?[0m
> auth
use this link to request the access code:
http://127.0.0.1:45678/authorize?client_id=fdea9ae7b29147fa98559b2030212918&redirect_uri=http://localhost:8080&response_type=code
waiting for code...
Tester: making requests to redirect uri: http://localhost:8080
Tester: Error: Connection refused
making http request for access_token...
response:
{"access_token":"456456","token_type":"Bearer","expires_in":3600,"refresh_token":"567567","scope":""}

---SUCCESS---
> categories
Top Lists

Super Mood

Party Time

> exit
---GOODBYE!---
?[1;31m
Start test 3?[0m
> categories
Please, provide access for application.
> exit
---GOODBYE!---
?[1;31m
Start test 4?[0m
> auth
use this link to request the access code:
http://127.0.0.1:45678/authorize?client_id=fdea9ae7b29147fa98559b2030212918&redirect_uri=http://localhost:8080&response_type=code
waiting for code...
Tester: making requests to redirect uri: http://localhost:8080
Tester: Error: Connection refused
making http request for access_token...
response:
{"access_token":"456456","token_type":"Bearer","expires_in":3600,"refresh_token":"567567","scope":""}

---SUCCESS---
> featured
Monday Morning Mood
http://open.spotify.com/user/spotify/playlist/6ftJBzU2LLQcaKefMi7ee7

Upp och hoppa!
http://open.spotify.com/user/spotify__sverige/playlist/4uOEx4OUrkoGNZoIlWMUbO

> exit
---GOODBYE!---
?[1;31m
Start test 5?[0m
> featured
Please, provide access for application.
> exit
---GOODBYE!---
?[1;31m
Start test 6?[0m
> auth
use this link to request the access code:
http://127.0.0.1:45678/authorize?client_id=fdea9ae7b29147fa98559b2030212918&redirect_uri=http://localhost:8080&response_type=code
waiting for code...
Tester: making requests to redirect uri: http://localhost:8080
Tester: Error: Connection refused
making http request for access_token...
response:
{"access_token":"456456","token_type":"Bearer","expires_in":3600,"refresh_token":"567567","scope":""}

---SUCCESS---
> new
Runnin'
[Pharrell Williams]
https://open.spotify.com/album/5ZX4m5aVSmWQ5iHAPQpT71

Sneakin'
[Drake2, Drake3]
https://open.spotify.com/album/0geTzdk2InlqIoB16fW9Nd

> exit
---GOODBYE!---
?[1;31m
Start test 7?[0m
> new
Please, provide access for application.
> exit
---GOODBYE!---
?[1;31m
Start test 8?[0m
> auth
use this link to request the access code:
http://127.0.0.1:45678/authorize?client_id=fdea9ae7b29147fa98559b2030212918&redirect_uri=http://localhost:8080&response_type=code
waiting for code...
Tester: making requests to redirect uri: http://localhost:8080
Tester: Error: Connection refused
making http request for access_token...
response:
{"access_token":"456456","token_type":"Bearer","expires_in":3600,"refresh_token":"567567","scope":""}

---SUCCESS---
> playlists Party Time
]]></system-out>
  <system-err><![CDATA[java.net.ConnectException: Connection refused
	at java.net.http/jdk.internal.net.http.HttpClientImpl.send(HttpClientImpl.java:561)
	at java.net.http/jdk.internal.net.http.HttpClientFacade.send(HttpClientFacade.java:119)
	at Server.run(Server.java:74)
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:779)
	at java.net.http/jdk.internal.net.http.PlainHttpConnection$ConnectEvent.handle(PlainHttpConnection.java:128)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.handleEvent(HttpClientImpl.java:957)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.lambda$run$3(HttpClientImpl.java:912)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.run(HttpClientImpl.java:912)
java.net.ConnectException: Connection refused
	at java.net.http/jdk.internal.net.http.HttpClientImpl.send(HttpClientImpl.java:561)
	at java.net.http/jdk.internal.net.http.HttpClientFacade.send(HttpClientFacade.java:119)
	at Server.run(Server.java:74)
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:779)
	at java.net.http/jdk.internal.net.http.PlainHttpConnection$ConnectEvent.handle(PlainHttpConnection.java:128)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.handleEvent(HttpClientImpl.java:957)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.lambda$run$3(HttpClientImpl.java:912)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.run(HttpClientImpl.java:912)
java.net.ConnectException: Connection refused
	at java.net.http/jdk.internal.net.http.HttpClientImpl.send(HttpClientImpl.java:561)
	at java.net.http/jdk.internal.net.http.HttpClientFacade.send(HttpClientFacade.java:119)
	at Server.run(Server.java:74)
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:779)
	at java.net.http/jdk.internal.net.http.PlainHttpConnection$ConnectEvent.handle(PlainHttpConnection.java:128)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.handleEvent(HttpClientImpl.java:957)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.lambda$run$3(HttpClientImpl.java:912)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.run(HttpClientImpl.java:912)
java.net.ConnectException: Connection refused
	at java.net.http/jdk.internal.net.http.HttpClientImpl.send(HttpClientImpl.java:561)
	at java.net.http/jdk.internal.net.http.HttpClientFacade.send(HttpClientFacade.java:119)
	at Server.run(Server.java:74)
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:779)
	at java.net.http/jdk.internal.net.http.PlainHttpConnection$ConnectEvent.handle(PlainHttpConnection.java:128)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.handleEvent(HttpClientImpl.java:957)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.lambda$run$3(HttpClientImpl.java:912)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.run(HttpClientImpl.java:912)
java.net.ConnectException: Connection refused
	at java.net.http/jdk.internal.net.http.HttpClientImpl.send(HttpClientImpl.java:561)
	at java.net.http/jdk.internal.net.http.HttpClientFacade.send(HttpClientFacade.java:119)
	at Server.run(Server.java:74)
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:779)
	at java.net.http/jdk.internal.net.http.PlainHttpConnection$ConnectEvent.handle(PlainHttpConnection.java:128)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.handleEvent(HttpClientImpl.java:957)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.lambda$run$3(HttpClientImpl.java:912)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.run(HttpClientImpl.java:912)
]]></system-err>
</testsuite>
